modules/seqexec/server/src/main/scala/seqexec/server/SeqTranslate.scala

```diff
@@ -226,95 +225,49 @@ class SeqTranslate(site: Site, systems: Systems[IO], settings: TranslateSettings
     deliverObserveCmd(seqId, f)
   }
 
-  private def pausedCommand(seqId: Observation.Id,
-                            f: ObserveControl[IO] => Option[Time => IO[ObserveCommandResult]],
-                            useCountdown: Boolean)(
+  def resumePaused(seqId: Observation.Id)(
     implicit cio: Concurrent[IO],
              tio: Timer[IO]
-  ): EngineState => Option[Stream[IO, executeEngine.EventType]] = st => {
-
-    def resumeIO(c: ObserveContext[IO], resumeCmd: IO[ObserveCommandResult]): Stream[IO, Result[IO]] =
-      for {
-        r <- Stream.eval(resumeCmd)
-        ret <- c.t(r)
-      } yield ret
-
-    def seqCmd(seqState: Sequence.State[IO], instrument: Instrument): Option[Stream[IO,
-      executeEngine.EventType]] = {
-
-      val inst = toInstrumentSys(instrument).toOption
-
-      val observeIndex: Option[(ObserveContext[IO], Option[Time], Int)] =
-        seqState.current.execution.zipWithIndex.find(_._1.kind === ActionType.Observe).flatMap {
+  ): EngineState => Option[Stream[IO, executeEngine.EventType]] = (st: EngineState) => {
+    val observeIndex: Option[(ObserveContext[IO], Option[Time], Int)] =
+      st.sequences.get(seqId)
+        .flatMap(_.seq.current.execution.zipWithIndex.find(_._1.kind === ActionType.Observe).flatMap {
```

Comment:
One thing here. When I was doing N&S I did it with multiple `Observe` actions and was hit by this `find` which only returns one.
At the end it was not a problem because for other reasons we had to revert to a single `Observe` action but the underlying assumption of a single action remains.
Could/Should we generalize this?

Reply:
I am not sure. The premise here is that for each step there is one observe and one image.

Reply:
I don't think we need to fix it now, but we need to be aware of this, perhaps a comment?